### ISSUE TITLE
[Bug?]: Cannot read request body in middleware or api route using cloudflare

### ISSUE THREAD

**Issue by sabercoy** [solidjs/solid-start#1287]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

/routes/api/test.ts
```typescript
import { APIEvent, readBody} from "@solidjs/start/server";

export const POST = async (event: APIEvent) => {
  const body = await readBody(event)

  return new Response(body)
}

```

Wrangler gives the error when trying to parse the body, as well as cloudflare when actually deployed
```
✘ [ERROR] A hanging Promise was canceled. This happens when the worker runtime is waiting for a Promise from JavaScript to resolve, but has detected that the Promise cannot possibly ever resolve because all code and events related to the Promise's I/O context have already finished.


✘ [ERROR] Uncaught (in response) Error: The script will never generate a response.
```

### Expected behavior 🤔

give the parsed body and no error

### Steps to reproduce 🕹

Steps:

1. clone https://github.com/Numnumberry/solid-start-cf-body-error
2. pnpm i
3. pnpm run wrangler
4. send a POST request to localhost:3000/api/test

**Comment by Numnumberry** [comment]
found a clue: `readBody` uses `readRawBody` which uses `WritableStream`
according to cloudflare, you cannot use WritableStream constructor on workers platform: https://developers.cloudflare.com/workers/runtime-apis/streams/writablestream/#writablestream

**Comment by ryansolid** [comment]
Thanks for reporting. Stuff like this shouldn't impact how we want to approach our API. This should be handled at the runtime level. I will tag this appropriately and forward it on.

**Comment by ryansolid** [comment]
Question, does it work if you try to get it off the request.

Like:
```js
const body = await event.request.text()
```
The reason I ask is preferably we are pushing people towards that interface and it occurs to me that `readBody` might be grabbing from a non-web location which is causing issues. We've seen it elsewhere. But it seemed to have been resolved. I suppose this might be cloudflare specific.

**Comment by sabercoy** [comment]
> Question, does it work if you try to get it off the request.

`event.request.text()` fails in the originally-posted test case the same as readBody()
```
"@solidjs/meta": "^0.29.2",
"@solidjs/router": "^0.10.10",
"@solidjs/start": "^0.4.10",
"solid-js": "^1.8.12",
"vinxi": "^0.1.6"
```
However...

I pulled down the latest solid@latest and reran the test:
`event.request.text()` succeeds
`event.request.json()` succeeds
`readBody(event)` succeeds
```
"@solidjs/meta": "^0.29.2",
"@solidjs/router": "^0.12.0",
"@solidjs/start": "^0.5.5",
"solid-js": "^1.8.14",
"vinxi": "^0.2.1"
```

It seems it works locally on wrangler!

**Comment by sabercoy** [comment]
wait there is no readBody in this version, I must have tested the wrong server, let me try again

**Comment by sabercoy** [comment]
yeah .text() and .json() both seems to work, I guess this is the expected way to go now?

**Comment by nksaraf** [comment]
I guess we can close this then. Please reopen if the issue persists with the latest solid-start and solid versions
